main.go

```diff
@@ -111,69 +185,117 @@ func checkShellyUpdateStatus(hostname string) (*shellyUpdateStatusResponse, erro
 	return makeShellyUpdateRequest(hostname, false)
 }
 
-func updateShelly(instance *zeroconf.ServiceEntry, wg *sync.WaitGroup) {
-	defer wg.Done()
-
-	shellyAddress := instance.AddrIPv4[0].String()
-
+func updateShellyGen1(name, address string) {
 	// First, we trigger a check for updates
-	fmt.Printf("[%s] checking for updates...\n", instance.HostName)
-	_, err := triggerShellyUpdateCheck(shellyAddress)
+	fmt.Printf("[%s,%s] checking for updates (gen1)...\n", name, address)
```

Comment:
Maybe we can create a `prefix` and add it to all `Printf` commands? Same for `updateShellyGen2`

```suggestion
        prefix := fmt.Sprintf("%s/%s/gen1", name, address)
	fmt.Printf("[%s] checking for updates...\n", prefix)
```

Reply:
Done.